qiskit/pulse/transforms/alignments.py

```diff
@@ -46,29 +44,53 @@ def align(self, schedule: Schedule) -> Schedule:
         """
         pass
 
+    @deprecated_functionality
     def to_dict(self) -> Dict[str, Any]:
         """Returns dictionary to represent this alignment."""
         return {"alignment": self.__class__.__name__}
 
+    @deprecated_functionality
+    @property
+    def is_sequential(self):
+        """Deprecated attribute.
+
+        Use superclass :class:`.SequentialContext` or :class:`.ParallelContext`.
+        """
+        return None
+
     def __eq__(self, other):
         """Check equality of two transforms."""
-        return isinstance(other, type(self)) and self.to_dict() == other.to_dict()
+        if type(self) is not type(other):
+            return False
+        if self._context_params != other._context_params:
+            return False
+        return True
 
     def __repr__(self):
-        name = self.__class__.__name__
-        opts = self.to_dict()
-        opts.pop("alignment")
-        opts_str = ", ".join(f"{key}={val}" for key, val in opts.items())
-        return f"{name}({opts_str})"
+        return f"{self.__class__.__name__}({', '.join(self._context_params)})"
+
+
+class SequentialContext(AlignmentKind):
+    """Alignment class that aligns all instructions in different channels in sequential."""
+
+    pass
+
+
+class ParallelContext(AlignmentKind):
+    """Alignment class that aligns all instructions in different channels in parallel."""
+
+    pass
```

Comment:
This makes me think we can just turn `is_sequential` into abstract method, rather than preparing a class. Perhaps this is much cleaner. Done in https://github.com/Qiskit/qiskit-terra/pull/7300/commits/48c70fe458ccd0371b5e3993ee5263aa62c00b36